docs/rules/import-path-order.md

```diff
@@ -0,0 +1,272 @@
+import a from './a';
+import b from 'b';
+```
+
+```js
+import a from './a';
+import b from '../b';
+```
+
+```js
+import a from '../a';
+import b from '../../b';
+```
+
+```js
+// Built-in packages should come first
+import a from 'a';
+import fs from 'fs';
+```
+
+## Pass
+
+```js
+const a = require('a');
+const b = require('b');
+```
+
+```js
+import a from 'a';
+import b from 'b';
+```
+
+```js
+import z from 'a';
+import y from 'b';
+import x from 'c';
+```
+
+```js
+import 'a';
+import b from 'b';
+```
+
+```js
+import b from 'b';
+import a from './a';
+```
+
+```js
+import b from '../b';
+import a from './a';
+```
+
+```js
+import b from '../../b';
+import a from '../a';
+```
+
+```js
+// Built-in packages should come first
+import fs from 'fs';
+import a from 'a';
+```
+
+## Options
+
+### `allowBlankLines`
+
+Type: `boolean`<br>
+Default: `false`
+
+Allow blank lines between import statements.
+
+#### Fail
+
+```js
+/* eslint unicorn/import-path-order: ["error", {allowBlankLines: false}] */
+import b from 'b';
+
+import a from 'a';
+```
+
+#### Pass
+
+```js
+/* eslint unicorn/import-path-order: ["error", {allowBlankLines: true}] */
+import a from 'a';
+
+import b from 'b';
+```
+
+```js
+/* eslint unicorn/import-path-order: ["error", {allowBlankLines: false}] */
+import a from 'a';
+// Comments are okay
+import b from 'b';
+```
+
+```js
+/* eslint unicorn/import-path-order: ["error", {allowBlankLines: false}] */
+import a from 'a';
+const foo = a.foo; // Non-import statements will fail
+import b from 'b';
+```
+
+### `comparator`
+
+Type: `string`
+
+This adds additional ordering constraints for imports that have the same relative source.
+
+Options are:
+- `case-sensitive` - Orders alphabetically, case-sensitive
+- `case-insensitive` - Orders alphabetically, case-insensitive
+- `parts` - Groups imports with similar parts next to each other *(default)*
+- `off` - No additional ordering
+
+#### Fail
+
+```js
+/* eslint unicorn/import-path-order: ["error", {comparator: 'case-sensitive'}] */
+import a from 'a';
+import B from 'B';
+```
+
+```js
+/* eslint unicorn/import-path-order: ["error", {comparator: 'case-insensitive'}] */
+import B from 'B';
+import a from 'a';
+```
+
+```js
+/* eslint unicorn/import-path-order: ["error", {comparator: 'parts'}] */
+import a from 'a-one';
+import b from 'b-three';
+import a from 'a-two';
+```
+
+#### Pass
+
+```js
+/* eslint unicorn/import-path-order: ["error", {comparator: 'case-sensitive'}] */
+import B from 'B';
+import a from 'a';
+```
+
+```js
+/* eslint unicorn/import-path-order: ["error", {comparator: 'case-insensitive'}] */
+import a from 'a';
+import B from 'B';
+```
+
+```js
+/* eslint unicorn/import-path-order: ["error", {comparator: 'parts'}] */
+import a from 'a-one';
+import a from 'a-two';
+import b from 'b-three';
+```
+
+```js
+/* eslint unicorn/import-path-order: ["error", {comparator: 'off'}] */
+import c from 'a-c';
+import b from 'b';
+import a from 'a';
+```
+
+### `partsRegex`
```

Comment:
Do we really need this option? What use-cases are there for it?

Reply:
I habitually expose anything that configures how code works. We can hide it if you want but the code needs to know what part separators are so I put it in the config.

The use case would be if someone doesn't want `/` or `-` to matter (or perhaps wants to also treat `_` as a separator.)

Reply:
I generally only expose things when I can think of valid use-cases that many will have, otherwise I wait until someone requests it.

> The use case would be if someone doesn't want / or - to matter

If that's the only use-case, there's probably a simpler way to expose it. Not really sure if even that is valid though. I cannot think of when I would want to individually control them.

> or perhaps wants to also treat _ as a separator.

`_` is not a valid package name, so it would only apply to local filenames, and I cannot really think of any good reason using that.

Reply:
Alright, I'll hide the option. It's trivial to reintroduce if someone asks for it.